types/pipeline_events.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import "time"
+
+type PipelineEventSystem string
+
+const (
+	PipelineEventSystemDeployment PipelineEventSystem = "k8s:deployment"
+	PipelineEventSystemPod        PipelineEventSystem = "k8s:pod"
+)
+
+// PipelineEvent model.
+type PipelineEvent struct {
+	PipelineID string              `json:"pipelineID" yaml:"pipelineID"`
+	System     PipelineEventSystem `json:"system" yaml:"system"`
+	Status     PipelineStatusKind  `json:"kind" yaml:"kind"`
+	Reason     string              `json:"reason" yaml:"reason"`
+	Message    string              `json:"message" yaml:"message"`
+	CreatedAt  time.Time           `json:"createdAt" yaml:"createdAt"`
```

Comment:
Normally, `createdAt` is a field generated by cloud.
So it's not present on the request bodies.

Unless, it is the time the message was created on kubernetes. In that case I would rename to avoid confusion.

Reply:
> Unless, it is the time the message was created on kubernetes. In that case I would rename to avoid confusion.

That is the idea. Should I name it simply `Time` or `LogTime` or `LocalTime` or `ClusterTime`?

Reply:
I've settled on EventTime for now..